src/test/compile-fail/issue-39553.rs

```diff
@@ -0,0 +1,23 @@
+// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![feature(conservative_impl_trait)]
+
+struct FragmentKey;
+struct Fragment;
+
+pub fn iter() -> impl Iterator<Item=(&FragmentKey, &Box<Fragment>)> {
```

Comment:
Ah. I will remove this test case tomorrow then and unmark the issue as E-needstest.